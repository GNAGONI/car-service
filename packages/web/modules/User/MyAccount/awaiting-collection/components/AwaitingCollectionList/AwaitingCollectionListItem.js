import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'server/pages';
import Button from 'react-bootstrap/Button';

// Constants
import { iconVehiclesSvg } from 'static/images/icons';

const AwaitingCollectionListItem = item => {
  const { id, imgUrl, carMake, carModel, year, currentMaxBid, locationFar, location } = item;

  const detailsLink = {
    pathname: '/my-account/awaiting-collection/collection-details',
    query: { id },
  };

  return (
    <Fragment key={id}>
      <div className="row__item live-list__item">
        <div className="product__detail">
          <div className="product-image">
            {imgUrl ? (
              <Link route={detailsLink}>
                <a>
                  <img className="img-fluid" src={imgUrl} alt="awaiting collection" />
                </a>
              </Link>
            ) : (
              <div className="awaiting-box">
                <img className="img-fluid" src={iconVehiclesSvg} alt="awaiting collection" />
                <span className="awaiting-text">Awaiting Image</span>
              </div>
            )}
          </div>

          <div className="text-holder">
            <Link route={detailsLink}>
              <a>
                {year} {carMake} <br /> {carModel}
              </a>
            </Link>
          </div>
        </div>
      </div>
      <div className="row__item live-list__item live-list__item-pl">
        <div className="price">Â£{currentMaxBid}</div>
      </div>
      <div className="row__item live-list__item live-list__item-pl">
        <address>
          {location} <span className="text-count">({locationFar})</span>
        </address>
      </div>
      <div className="row__item live-list__item live-list__item-pl">
        <Link route={detailsLink}>
          <Button variant="success">View details</Button>
        </Link>
      </div>
    </Fragment>
  );
};

AwaitingCollectionListItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number,
    carMake: PropTypes.string,
    carModel: PropTypes.string,
    numberOfBids: PropTypes.number,
    currentMaxBid: PropTypes.string,
    timeToEnd: PropTypes.number,
    watched: PropTypes.bool,
    imgUrl: PropTypes.string,
    year: PropTypes.number,
    location: PropTypes.string,
    locationFar: PropTypes.string,
  }),
};

export default AwaitingCollectionListItem;
