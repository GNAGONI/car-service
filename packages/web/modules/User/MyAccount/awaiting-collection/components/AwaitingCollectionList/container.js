import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { compose } from 'redux';
import {
  getCarsList,
  getCarsListPaging,
  isCarListLoaded,
  carListErrorMessage,
} from '@car-service/redux/selectors/awaitingCollection';
import { getAwaitingCollectionCarListRequest } from '@car-service/redux/actions/awaitingCollection';
import AwaitingCollectionList from 'modules/User/MyAccount/awaiting-collection/components/AwaitingCollectionList';
import CheckDataError from 'components/checkDataError';
import showItemsList from 'components/showItemsList';
import checkApiError from 'components/checkApiError';

const mapStateToProps = createStructuredSelector({
  items: getCarsList(),
  paging: getCarsListPaging(),
  isLoaded: isCarListLoaded,
  errorMessage: carListErrorMessage,
});

const mapDispatchToProps = dispatch => ({
  getCarList: data => dispatch(getAwaitingCollectionCarListRequest(data)),
});

export default compose(
  connect(
    mapStateToProps,
    mapDispatchToProps,
  ),
  showItemsList,
  component => checkApiError(component, CheckDataError),
)(AwaitingCollectionList);
