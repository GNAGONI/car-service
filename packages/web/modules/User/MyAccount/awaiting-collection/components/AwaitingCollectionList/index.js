import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

// Components
import List from 'components/list';
import LoadMore from 'components/loadMore';
import AwaitingCollectionListItem from 'modules/User/MyAccount/awaiting-collection/components/AwaitingCollectionList/AwaitingCollectionListItem';

const awaitingCollectionListHeader = [
  {
    label: 'Car Details',
    key: 'carDetails',
  },
  {
    label: 'Winning Bid',
    key: 'winningBid',
  },
  {
    label: 'Location',
    key: 'location',
  },
  {
    label: 'Actions',
    key: 'actions',
  },
];

class AwaitingCollectionList extends PureComponent {
  static defaultProps = {
    items: [],
    getCarList: () => {},
    paging: {},
  };

  static propTypes = {
    getCarList: PropTypes.func,
    paging: PropTypes.shape({
      offset: PropTypes.number,
      limit: PropTypes.number,
      totalNumberOfAuctions: PropTypes.number,
    }),
    items: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.number,
        carMake: PropTypes.string,
        carModel: PropTypes.string,
        numberOfBids: PropTypes.number,
        currentMaxBid: PropTypes.string,
        timeToEnd: PropTypes.number,
        watched: PropTypes.bool,
        imgUrl: PropTypes.string,
        year: PropTypes.number,
        location: PropTypes.string,
        locationFar: PropTypes.string,
      }),
    ),
  };

  componentDidMount() {
    const { getCarList, paging, items } = this.props;

    if (!items.length) {
      getCarList({ ...paging });
    }
  }

  onLoadMore = data => {
    const { getCarList, paging } = this.props;

    getCarList({ ...paging, ...data });
  };

  render() {
    const {
      items,
      paging: { offset, limit, totalNumberOfAuctions },
    } = this.props;

    return (
      <div>
        {
          <LoadMore
            onLoadMore={this.onLoadMore}
            offset={offset}
            limit={limit}
            isShowButton={items.length !== totalNumberOfAuctions}
          >
            <div className="awaiting-collection__list">
              <List
                headers={awaitingCollectionListHeader}
                items={items}
                keyProp="id"
                renderer={AwaitingCollectionListItem}
              />
            </div>
          </LoadMore>
        }
      </div>
    );
  }
}

export default AwaitingCollectionList;
