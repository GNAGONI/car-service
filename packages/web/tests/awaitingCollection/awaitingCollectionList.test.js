import React from 'react'
import renderer from 'react-test-renderer'
import AwaitingCollectionList from 'components/awaitingCollection/awaitingCollectionList'

it('Renders correctly', () => {
  const tree = renderer.create(<AwaitingCollectionList />).toJSON()

  expect(tree).toMatchSnapshot()
})
