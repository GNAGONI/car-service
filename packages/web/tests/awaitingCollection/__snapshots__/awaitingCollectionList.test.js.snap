// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div
    className="load-more"
  >
    <div
      className="load-more__content mb-5"
    >
      <div
        className="awaiting-collection__list"
      >
        <section
          className="list"
        >
          <div
            className="list__row list__row-header"
          >
            <div
              className="row__item row__item-header"
            >
              <strong
                className="data-title"
              >
                Car Details
              </strong>
            </div>
            <div
              className="row__item row__item-header"
            >
              <strong
                className="data-title"
              >
                Winning Bid
              </strong>
            </div>
            <div
              className="row__item row__item-header"
            >
              <strong
                className="data-title"
              >
                Location
              </strong>
            </div>
            <div
              className="row__item row__item-header"
            >
              <strong
                className="data-title"
              >
                Actions
              </strong>
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      className="load-more__footer text-center"
    >
      <button
        className="btn btn-new btn-success"
        onClick={[Function]}
      >
        Load more
      </button>
    </div>
  </div>
</div>
`;
